pyproject.toml

```diff
@@ -76,13 +76,31 @@ test = [
 # Dependencies needed to run the full test suite
 ci = [
   "ipympl",  # needed for notebook testing
+  "jax",
+  "matplotlib",
   "mpi-pytest",
   "nbval",
+  "ngsPETSc",
   "pylit",
   "pytest",
   "pytest-split",  # needed for firedrake-run-split-tests
   "pytest-timeout",
   "pytest-xdist",
+  "slepc4py",
+  "torch",  # requires passing '--extra-index-url' to work
```

Comment:
Is this going to stop us putting firedrake on PyPI? I think they don't allow off-site dependencies. Not sure if that rule extends to optional dependencies.

Reply:
I don't think so. I think we just need to tell users to pass an extra flag to `pip install` if they want to install this set of optional dependencies.

More generally I am pretty unhappy with the growing number of optional dependencies/different configurations and would like to clean it up at some point.